io_handler_mpi.f90

```diff
@@ -1,10 +1,33 @@
 #include "errors.fpp"
 
+#define MPI_WRAPPER(function, handle, obj, size, mytype, status, err) \
+select type(obj); \
+    type is (integer(ik)); \
+      call function(handle, obj, size, mytype, status, err); \
+    type is (integer(hik)); \
+      call function(handle, obj, size, mytype, status, err); \
+    type is (real); \
+      call function(handle, obj, size, mytype, status, err); \
+    type is (real(rk)); \
+      call function(handle, obj, size, mytype, status, err); \
+    type is (real(ark)); \
+      call function(handle, obj, size, mytype, status, err); \
+    type is (complex); \
+      call function(handle, obj, size, mytype, status, err); \
+    type is (character(len=*)); \
+      call function(handle, obj, size, mytype, status, err); \
+    class default; \
+      write(*,*) 'Not covered'; \
+end select
+
+
 module io_handler_mpi
+#ifdef TROVE_USE_MPI_
```

Comment:
Good point, I'll remove it.

Reply:
Good point, I'll remove it.

Reply:
Actually, we do need it currently because we're listing `io_handler_mpi.o` as a dependency of `perturbation.o`, so we always compile it. We could make that rule conditional though.